api.py

```diff
@@ -0,0 +1,277 @@
+import json
+import tempfile
+import uuid
+from datetime import datetime
+from tools import DatabaseConnector
+from flask import Flask, request, jsonify, send_file
+from scrapy.crawler import CrawlerRunner
+from scrapy.utils.project import get_project_settings
+from wsf_scraping.spiders import who_iris_spider, nice_spider
+from twisted.internet import reactor, endpoints
+from scrapy.utils.log import configure_logging
+from twisted.web import server, wsgi
+
+
+class ScrAPI(Flask):
+
+    def __init__(self, import_name=__package__, **kwargs):
+            super(ScrAPI, self).__init__(import_name, **kwargs)
+            configure_logging({'LOG_FORMAT': '%(levelname)s: %(message)s'})
+            self._init_url_rules()
+            self.process = CrawlerRunner(get_project_settings())
+            self.tp = reactor.getThreadPool()
+            self.database = DatabaseConnector()
+            self.response_meta = {"meta": {
+                "project": "WSF Web Scraper"
+            }}
+
+    def __del__(self):
+        self.database._close_all_spiders()
+        self.database.cursor.close()
+        self.database.connection.close()
+
+    def run(self, host=None, port=None, debug=None, **options):
+        super(ScrAPI, self).run(host, port, debug, **options)
+
+    def _get_meta_response(self, res):
+        res.update(self.response_meta)
+        return res
+
+    def _init_url_rules(self):
+        """Attach the endpoints to run spiders and list the spiders
+        that are available in the API
+        """
+
+        self.add_url_rule(
```

Comment:
Nice 👍 Much better :)

Reply:
for future reference, this is a nice flask extension for rest API that I ve used in the past and is quite handy. https://flask-restful.readthedocs.io/en/latest/